src/linear_algebra/finite_dimensional.lean

```diff
@@ -1022,6 +1026,25 @@ lemma set_is_basis_of_linear_independent_of_card_eq_findim
   is_basis K (coe : s → V) :=
 is_basis_of_linear_independent_of_card_eq_findim lin_ind (trans s.to_finset_card.symm card_eq)
 
+lemma findim_one_implies_equal {F E : Type*} [field F] [field E] [algebra F E]
+  (E_dim : findim F E = 1) : (⊤ : subalgebra F E) = ⊥ :=
+begin
+  ext,
+  apply iff_of_true algebra.mem_top,
+  set s : set E := {1} with hs,
+  have : fintype s := unique.fintype,
+  have s_lin_ind : linear_independent F (coe : s → E) := linear_independent_singleton one_ne_zero,
+  have s_card : s.to_finset.card = findim F E := by { change s.to_finset.card with 1, rw E_dim },
+  obtain ⟨_, s_spans⟩ := set_is_basis_of_linear_independent_of_card_eq_findim s_lin_ind s_card,
+  have x_in_span_one : x ∈ submodule.span F s,
+  { rw subtype.range_coe at s_spans,
+    rw s_spans,
+    exact submodule.mem_top, },
+  obtain ⟨a, ha⟩ := submodule.mem_span_singleton.mp x_in_span_one,
+  rw algebra.mem_bot,
+  exact ⟨a, by rw [← ha, algebra.smul_def, mul_one]⟩,
+end
```

Comment:
Here is a slight reworking using `eq_bot_iff`. It might be nicer for end users to do this in two steps: show that a subalgebra is `⊥` iff its dimension is 1, then show that the dimension of `⊤` is the dimension of `1`.

```suggestion
begin
  rw eq_bot_iff,
  rintro x -,
  rw [subalgebra.mem_coe, algebra.mem_bot],
  set s : set E := {1} with hs,
  have : fintype s := unique.fintype,
  have s_lin_ind : linear_independent F (coe : s → E) := linear_independent_singleton one_ne_zero,
  have s_card : s.to_finset.card = findim F E := by { change s.to_finset.card with 1, rw E_dim },
  obtain ⟨_, s_spans⟩ := set_is_basis_of_linear_independent_of_card_eq_findim s_lin_ind s_card,
  have x_in_span_one : x ∈ submodule.span F s,
  { rw subtype.range_coe at s_spans,
    rw s_spans,
    exact submodule.mem_top, },
  obtain ⟨a, ha⟩ := submodule.mem_span_singleton.mp x_in_span_one,
  exact ⟨a, by rw [← ha, algebra.smul_def, mul_one]⟩,
end
```

Reply:
I ended up substantially refactoring this and adding a bunch of little lemmas. Let me know if there are problems.